src/pybind/mgr/dashboard/services/sso.py

```diff
@@ -46,17 +48,19 @@ class SsoDB(object):
     VERSION = 1
     SSODB_CONFIG_KEY = "ssodb_v"
 
-    def __init__(self, version, protocol, saml2):
+    def __init__(self, version, protocol, saml2, oauth2):
```

Comment:
As it doesn't make sense (and it'll probably not work) to have both SAML2 and OAuth2 together:
```suggestion
    def __init__(self, version, protocol: Literal["saml2", "oauth2"], config: Union[SAML2Config, OAuth2Config]):
```